onnxruntime/core/providers/cpu/tensor/onehot.cc

```diff
@@ -28,19 +28,35 @@ namespace onnxruntime {
 // spec: https://github.com/onnx/onnx/blob/master/docs/Operators.md#OneHot
 
 // T1: indices, T2: depth, T3: values
-#define REG_TYPED_ONE_HOT_OP(types_str, in_type, out_type, depth_type)     \
+#define REG_TYPED_ONE_HOT_OP_V9_10(types_str, in_type, out_type, depth_type) \
+  ONNX_CPU_OPERATOR_VERSIONED_TYPED_KERNEL(                                  \
+      OneHot,                                                                \
+      9, 10,                                                                 \
+      types_str,                                                             \
+      KernelDefBuilder()                                                     \
+          .TypeConstraint("T1", DataTypeImpl::GetTensorType<in_type>())      \
+          .TypeConstraint("T2", DataTypeImpl::GetTensorType<depth_type>())   \
+          .TypeConstraint("T3", DataTypeImpl::GetTensorType<out_type>()),    \
+      OneHotOp<in_type, out_type, depth_type>);
+
+// T1: indices, T2: depth, T3: values
+#define REG_TYPED_ONE_HOT_OP_V11(types_str, in_type, out_type, depth_type) \
   ONNX_CPU_OPERATOR_TYPED_KERNEL(                                          \
       OneHot,                                                              \
-      9,                                                                   \
+      11,                                                                  \
       types_str,                                                           \
       KernelDefBuilder()                                                   \
           .TypeConstraint("T1", DataTypeImpl::GetTensorType<in_type>())    \
           .TypeConstraint("T2", DataTypeImpl::GetTensorType<depth_type>()) \
           .TypeConstraint("T3", DataTypeImpl::GetTensorType<out_type>()),  \
       OneHotOp<in_type, out_type, depth_type>);
 
-#define REG_ONE_HOT_OP(in_type, out_type, depth_type) \
-  REG_TYPED_ONE_HOT_OP(in_type##_##out_type##_##depth_type, in_type, out_type, depth_type)
+#define REG_ONE_HOT_OP(in_type, out_type, depth_type)                                             \
+  REG_TYPED_ONE_HOT_OP_V9_10(in_type##_##out_type##_##depth_type, in_type, out_type, depth_type); \
+  REG_TYPED_ONE_HOT_OP_V11(in_type##_##out_type##_##depth_type, in_type, out_type, depth_type)
+
+#define REG_ONE_HOT_OP_V11(in_type, out_type, depth_type) \
```

Comment:
I guess this macro is not needed anymore `REG_ONE_HOT_OP_V11` given that the above macro has been expanded to include V11 as well ?